### ISSUE TITLE
ofNode global matrix caching

### ISSUE THREAD

**Issue by chuckleplant** [openframeworks/openFrameworks#4205]
I went ahead and coded ofNode global matrix caching. Before PR-ing I wanted to discuss whether it's a good approach. You can see the changes [here](https://github.com/chuckleplant/openFrameworks/blob/feature_node_caching/libs/openFrameworks/3d/ofNode.cpp) in a feature branch. 

What this change does is computing the global matrix only if the local matrix changes. To do so every node keeps not only a parent ofNode* but also a set of children. This way we can flag all children as "dirty" whenever the node's matrix changes.

I've used a vector to make iteration faster, and a map to relate vector indices to the ofNode*. Tried first with an unordered_set<ofNode*> but access was slower. 

I'll do a clean PR when it's polished!

**Comment by bilderbuchi** [comment]
ping @openframeworks/2d-3d

**Comment by arturoc** [comment]
isn't it easier to just keep a dirty flag for each node's local transformation? when a node needs it's global it just transverses the hierarchy checking if anything is dirty, if it is it recalculates the parent transform otherwise it uses the cached global. having so many collections to transverse seems kind of complicated for something that probably won't optimize a lot i think

**Comment by chuckleplant** [comment]
Yes you're right. No need to keep all children nodes listed. I'll give it a try.
